### ISSUE TITLE
Dropdowns are not accessible with VoiceOver when contained within a Modal/Dialog

### ISSUE THREAD

**Issue by Hadeon** [microsoft/fluentui#17910]
### Environment Information

- **Package version(s)**: 8.11.1
- **Browser and OS versions**: iOS 14.4.2

### Describe the issue:

When you have a Dropdown contained within a Modal, if you open the Dropdown with iOS VoiceOver focus remains on the Modal and surrounding elements and cannot reach the slide out to select an option from the Dropdown

### Please provide a reproduction of the issue in a codepen:

https://u8j6d.csb.app/

#### Actual behavior:

When a Dropdown within a Modal is opened, focus remains on the Modal and the Dropdown options are unreachable.

#### Expected behavior:

When a Dropdown is opened focus should be moved to the Dropdown slideout so that selections can be made.
